src/components/datePicker/datePicker.tsx

```diff
@@ -0,0 +1,126 @@
+import ReactDatePicker, { registerLocale } from 'react-datepicker';
+import classNames from 'classnames/bind';
+import { Locale } from 'date-fns';
+import { FC, ReactNode, useRef, ReactElement } from 'react';
+import FieldText from '@components/fieldText';
+import { CalendarIcon } from '@/components/icons';
+import { DatePickerHeader } from './header/datePickerHeader';
+import styles from './datePicker.module.scss';
+
+const cx = classNames.bind(styles);
+
+const DEFAULT_LANGUAGE = 'en';
+const DEFAULT_DATE_FORMAT = 'MM-dd-yyyy';
+
+interface DatePickerProps {
+  onChange?: (date: Date | any) => void;
+  onBlur?: () => void;
+  onFocus?: () => void;
+  headerNodes?: ReactNode;
+  disabled?: boolean;
+  shouldCloseOnSelect?: boolean;
+  fixedHeight?: boolean;
+  startDate?: Date | undefined;
+  endDate?: Date | undefined;
+  customClassName?: string;
+  popperClassName?: string;
+  calendarClassName?: string;
+  customTimeInput?: ReactElement;
+  language?: string;
+  yearsOptions?: number[];
+  placeholder?: string;
+  dateFormat?: string;
+  selects?: 'start' | 'end' | string;
```

Comment:
What can be the `string` here? If it should have three states: 'start', 'end', 'unspecified' (or 'none' or smth like that), let's mention it here and not allow any string.